docs/content/languages/python.md

```diff
@@ -0,0 +1,356 @@
+
+### Accessing HTTP Requests
+
+#### HTTP Headers
+
+Write a simple `headers.py` with something like this:
+
+```python
+from flask import request
+
+def main():
+    myHeader = request.headers['x-my-header']
+    return "The header's value is '%s'" % myHeader
+```
+
+Create that function, assign it a route, and invoke it with an HTTP header:
+
+```bash
+$ fission function create --name headers --env python --code headers.py
+
+$ fission route create --name /headers --function headers
+
+$ curl -H "X-My-Header: Hello" $FISSION_ROUTER/headers 
+The header's value is 'Hello'
+```
+
+#### Query parameters
+
+HTTP Query parameters are the key-value pairs in a URL after the `?`.
+They are also available through the request object:
+
+Write a simple `query.py` with something like this:
+
+```python
+from flask import request
+
+def main():
+    queryParam = request.args.get('myKey')
+    return "Value for myKey: %s" % queryParam
+```
+
+Create that function, assign it a route, and invoke it with a query parameter:
+
+```bash
+$ fission function create --name query --env python --code query.py
+
+$ fission route create --name /query --function query
+
+$ curl $FISSION_ROUTER/query?myKey=myValue
+Value for myKey: myValue
+```
+
+#### Body 
+
+HTTP POST and PUT requests can have a request body.  Once again, you
+can access this body through the request object.
+
+For requests with a JSON Content-Type, you can directly get a parsed
+object with `request.get_json()`
+[[docs]](http://flask.pocoo.org/docs/1.0/api/#flask.Request.get_json).  
+
+For form-encoded requests ( application/x-www-form-urlencoded), use
+`request.form.get('key')`
+[[docs]](http://flask.pocoo.org/docs/1.0/api/#flask.Request.form).
+
+For all other requests, use `request.data`
+[[docs]](http://flask.pocoo.org/docs/1.0/api/#flask.Request.data) to
+get the full request body as a string of bytes.
+
+You can find the full docs on the request object in [the flask
+docs](http://flask.pocoo.org/docs/1.0/api/#incoming-request-data).
+
+### Controlling HTTP Responses
+
+The simplest way to return a response is to return a string.  This
+implicitly says that your function succeeded with a status code of
+200; the returned string becomes the body.  However, you can control
+the response more closely using the Flask `response` object.
+
+#### Setting Response Headers
+
+```python
+import flask
+
+def main():
+    resp = flask.Response("Hello, world!")
+    resp.headers['X-My-Response-Header'] = 'Something'
+    return resp
+```
+
+#### Setting Status Codes 
+
+```python
+import flask
+
+def main():
+    resp = flask.Response("Hello, world!")
+    resp.status_code = 200
+    return resp
+```
+
+#### HTTP Redirects
+
+```python
+import flask
+
+def main():
+    r = flask.redirect('/new-url', code=303)
+    
+    # Optional; set this to False to force a relative URL redirect.
+    # Defaults to True, which converts the redirect to an absolute URL.
+    r.autocorrect_location_header = False
```

Comment:
Looks like related to this issue: https://github.com/fission/fission/issues/918
We can add some more explanation here.
```
# Optional; set this to False to force a relative URL redirect.
# Defaults to True, which converts the redirect to an absolute URL 
# with a private host address which is not accessible from external.
```